internal/object/object.go

```diff
@@ -115,14 +115,36 @@ func NewNamespace(name string) *corev1.Namespace {
 	}
 }
 
-// NewContour makes a Contour object using the provided ns/name
-// for the object's namespace and name.
-func NewContour(name, ns string) *operatorv1alpha1.Contour {
+// NewContour makes a Contour object using the provided ns/name for the
+// object's namespace/name, pubType for the network publishing type of
+// Envoy, and Envoy container ports 8080/8443.
+func NewContour(name, ns, specNs string, remove bool, pubType operatorv1alpha1.NetworkPublishingType) *operatorv1alpha1.Contour {
```

Comment:
don't have to do it now, but we probably want to move to an options struct before long as this function's parameter list gets longer and longer.

Reply:
Agreed. As you mention, let me refactor into an options struct in a follow-on PR.

Reply:
xref: https://github.com/projectcontour/contour-operator/issues/189